File: 2nd Year/Term 2/Data Structures & Algorithms/MS1/Old files (mistake)/Array/Pseudocodes/Sortandinsert.py
# Sort and Insert in Array - Inventory (MotorPH)

def insert_item(inventory, brand, model, quantity):
    """Insert new inventory item (append to array)."""
    inventory.append({"brand": brand, "model": model, "quantity": quantity})


def sort_by_brand(inventory):
    """Sort inventory by brand."""
    inventory.sort(key=lambda x: x["brand"])


def display(inventory):
    """Display inventory list."""
    if not inventory:
        print("  (empty)")
        return
    for item in inventory:
        print(f"  - {item['brand']} {item['model']}: {item['quantity']} units")


def main():
    # Start with empty array
    inventory = []

    # Insert new inventory items
    insert_item(inventory, "Yamaha", "NMAX 155", 5)
    insert_item(inventory, "Honda", "PCX", 5)
    insert_item(inventory, "Suzuki", "Burgman Street", 7)

    print("After inserts (unsorted):")
    display(inventory)

    # Sort inventory by brand
    sort_by_brand(inventory)
    print("\nAfter sort by brand:")
    display(inventory)

    # Insert one more, then sort again
    insert_item(inventory, "Kawasaki", "Ninja 125", 4)
    print("\nAfter one more insert (unsorted):")
    display(inventory)
    sort_by_brand(inventory)
    print("After sort by brand:")
    display(inventory)


if __name__ == "__main__":
    main()

File: 2nd Year/Term 2/Data Structures & Algorithms/MS1/Old files (mistake)/Array/Pseudocodes/Sortandinsert.java
package Array.Pseudocodes;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

/**
 * Sort and Insert in Array - Inventory (MotorPH)
 */
public class Sortandinsert {

    record InventoryItem(String brand, String model, int quantity) {}

    /** Insert new inventory item (append to array). */
    static void insertItem(List<InventoryItem> inventory, String brand, String model, int quantity) {
        inventory.add(new InventoryItem(brand, model, quantity));
    }

    /** Sort inventory by brand. */
    static void sortByBrand(List<InventoryItem> inventory) {
        inventory.sort(Comparator.comparing(InventoryItem::brand));
    }

    /** Display inventory list. */
    static void display(List<InventoryItem> inventory) {
        if (inventory.isEmpty()) {
            System.out.println("  (empty)");
            return;
        }
        for (InventoryItem item : inventory) {
            System.out.println("  - " + item.brand() + " " + item.model() + ": " + item.quantity() + " units");
        }
    }

    public static void main(String[] args) {
        // Start with empty array (ArrayList)
        List<InventoryItem> inventory = new ArrayList<>();

        // Insert new inventory items
        insertItem(inventory, "Yamaha", "NMAX 155", 5);
        insertItem(inventory, "Honda", "PCX", 5);
        insertItem(inventory, "Suzuki", "Burgman Street", 7);

        System.out.println("After inserts (unsorted):");
        display(inventory);

        // Sort inventory by brand
        sortByBrand(inventory);
        System.out.println("\nAfter sort by brand:");
        display(inventory);

        // Insert one more, then sort again
        insertItem(inventory, "Kawasaki", "Ninja 125", 4);
        System.out.println("\nAfter one more insert (unsorted):");
        display(inventory);
        sortByBrand(inventory);
        System.out.println("After sort by brand:");
        display(inventory);
    }
}
